lib/gateway/find-value-types.js

```diff
@@ -0,0 +1,54 @@
+const {
+  isInterfaceType,
+  isObjectType,
+  isInputType
+} = require('graphql')
+
+function duplicates (array) {
+  const uniq = array
+    .map((item) => {
+      return {
+        count: 1,
+        item
+      }
+    })
+    .reduce((a, b) => {
+      a[b.item] = (a[b.item] || 0) + b.count
+      return a
+    }, {})
+  return Object.keys(uniq).filter((a) => uniq[a] > 1)
+}
```

Comment:
Why can't we just put all names in a set instead?

Reply:
Because `allTypes` is an array of `GraphQLObjectType` and not strings. Using the `GraphQLObjectType` instead of strings is required for `isObjectType` etc. to work properly.